src/components/fields/DateField.js

```diff
@@ -39,110 +44,110 @@ class DateField extends AskField {
       { backgroundColor: this.props.theme.inputBackground }
     )
   }
-
   validateAndSave(options) {
     if (this.validate()) {
-      this.setState({ value: this.buildValue() });
+      this.setState({
+        value: this.buildValue()
+      });
     } else {
-      this.setState({ value: '--' });
+      this.setState({
+        value: '--'
+      });
     }
     this.update(options);
   }
+  validate() {
+    const isValid = this.isDateValid(),
+          completed = this.isCompleted();
 
-  // Interface methods
+    this.setState({
+      isValid,
+      completed
+    });
 
-  validate() {
-    let isValid = true, isCompleted = false;
-    isValid = this.dateIsValid();
-    isCompleted = !!this.state.value.length;
-    this.setState({ isValid: isValid, completed: isCompleted });
-    return !!this.props.wrapper.required ? isValid && isCompleted : isValid;
+    return !!this.props.wrapper.required ? isValid && completed : isValid;
```

Comment:
you don't need the double bang here, and since there's a lot going in this ternary, please consider adding some comments explaining what's going on.